python/taichi/lang/ast/ast_transformer.py

```diff
@@ -1136,7 +1137,13 @@ def build_static_for(ctx, node, is_grouped):
             if len(targets) != 1:
                 raise TaichiSyntaxError(f"Group for should have 1 loop target, found {len(targets)}")
             target = targets[0]
+            iter_time = 0
+            warn_already = False
             for value in impl.grouped(ndrange_arg):
+                iter_time += 1
+                if not warn_already and ti_unroll_limit and iter_time > ti_unroll_limit:
+                    warn_already = True
+                    warnings.warn(f"loop unrolling only support under {ti_unroll_limit}")
```

Comment:
`You are unrolling more than {ti_unroll_limit} iterations, so the compile time may be extremely long. You can use a non-static for loop if you want to decrease the compile time. You can disable this warning by setting ti.init(unrolling_limit=0).`

Reply:
You can use `warnings.warn_explicit` to specify the source code of the warning. See https://github.com/taichi-dev/taichi/blob/a4d246ba886108579a169a2a0429bad0a8f7a83d/python/taichi/lang/ast/ast_transformer.py#L470-L481

Reply:
ok, I add rule to fall back static_for to normal for, and when user set unroll_limit=0, disable the fallback, force use static_for.

Reply:
> You can use `warnings.warn_explicit` to specify the source code of the warning. See
> 
> https://github.com/taichi-dev/taichi/blob/a4d246ba886108579a169a2a0429bad0a8f7a83d/python/taichi/lang/ast/ast_transformer.py#L470-L481

ok, I add rule to fall back static_for to normal for, and when user set unroll_limit=0, disable the fallback, force use static_for.